src/compiler.js

```diff
@@ -40,20 +40,20 @@ module.exports.run = ({ prefix, favicons: options, logo, cache: cacheDirectory }
 
       delete compilation.assets[output];
 
-      waterfall(compilation, 'webapp-webpack-plugin-before-emit', result, (error, { html, assets } = {}) => {
-          if (error) {
-            return reject(error);
-          }
-
-          for (const { name, contents } of assets) {
-            compilation.assets[name] = {
-              source: () => contents,
-              size: () => contents.length,
-            };
-          }
-
-          return resolve(html);
+      waterfall(compilation, 'webapp-webpack-plugin-before-emit', result, (error, { html = '', assets = [] } = {}) => {
+        if (error) {
+          return reject(error);
         }
+
+        for (const { name, contents } of assets) {
+          compilation.assets[name] = {
+            source: () => contents,
+            size: () => contents.length,
+          };
+        }
+
+        return resolve(html);
```

Comment:
It looks like this commit [fixes unit tests](https://travis-ci.org/brunocodutra/webapp-webpack-plugin/builds/385880434), the only remaining failure is due to the fact `package-lock.json` is somehow outdated, could you generate it again?

Apart from that, the only missing bit are unit tests to cover the new feature.

Reply:
It will be possible that I could update webpack at v. 4.10.2 too?  Or only do I generate the lock.json?

Reply:
It seems that updating webpack is not relevant for this PR, so I believe it's not necessary to modify package-lock.json at all.

We can always open a new PR to update it separetedly.

Reply:
Ok, I will wait for any update. Have a good day.